econ/econ.go

```diff
@@ -18,79 +18,60 @@ func ComputeAgentData(
 	sdk poolstypes.PoolsSDK,
 	agentLiquidAssets *big.Int,
 	principal *big.Int,
-	minerIDs []address.Address,
+	aggMinerStats *mstat.MinerStats,
 	agentAddr common.Address,
 	tsk *types.TipSet,
 ) (*vc.AgentData, error) {
-	lapi, closer, err := sdk.Extern().ConnectLotusClient()
-	if err != nil {
-		return nil, err
-	}
-	defer closer()
-
 	// here we just work our way through the AgentData, computing each key
 	// TODO: could probably parellize this elegently to speed things up
 	data := &vc.AgentData{}
 
-	aggMinerStats, err := mstat.ComputeMinersStats(ctx, minerIDs, tsk, lapi)
-	if err != nil {
-		return nil, err
-	}
-
 	data.QaPower = aggMinerStats.QualityAdjPower
 
 	data.GreenScore = aggMinerStats.GreenScore
 
 	data.AgentValue = big.NewInt(0).Add(agentLiquidAssets, aggMinerStats.Balance)
 
-	data.LiveSectors = aggMinerStats.LiveSectors
-	data.FaultySectors = aggMinerStats.FaultySectors
-
 	/* ~~~~~ CollateralValue ~~~~~ */
 
-	// CV = AgentValue * terminationPenalty
-	data.CollateralValue = CollateralValue(data.AgentValue)
-
-	/* ~~~~~ ExpectedDailyFaultPenalties ~~~~~ */
-
-	// COULD REMOVE
-	data.ExpectedDailyFaultPenalties = aggMinerStats.PenaltyFaultPerDay
-
-	/* ~~~~~ ExpectedDailyRewards ~~~~~ */
+	ats, err := sdk.Query().AgentPreviewTerminationPrecise(ctx, agentAddr, tsk)
+	if err != nil {
+		return nil, err
+	}
+	// here we replace the ats.AgentAvailableBal with the AgentLiquidAssets passed in this call to compute the post-action liquidation value
```

Comment:
Confusing...available balance is the same as liquid balance just in this case liquid assets has the post-action value